### PR TITLE
[Localization] localize hatch egg message (+ egg move unlock)

### PR DESCRIPTION
**Pull request by EnochG1**
## What are the changes?
Add the option to localize message text when egg is hatched.

## Why am I doing these changes?
All text should be localizable and I don't want to show message with mixed language.
As-is images in Korean.

| egg hatched | egg move unlocked |
| --- | --- |
| image | image |

## What did change?
 - Change the code for localize
 - Added localization key in each language folders.
 - Translated of them to Korean.

### Screenshots/Videos

| Lang | egg hatched | egg move unlocked |
| :-----: | :-----: | :-----: |
| En | image | image |
| Ko | image | image |
| De | image | image |
| zh_CN | image | image |
| Fr | image | image |
| Es | image | image |

## How to test the changes?
Manually.
1. Checkout this PR and change language
2. set overrides.ts ```IMMEDIATE_HATCH_EGGS_OVERRIDE = true;``` and try hatch the egg.

## Checklist
- [X] There is no overlap with another PR?
- [X] The PR is self-contained and cannot be split into smaller PRs?
- [X] Have I provided a clear explanation of the changes?
- [X] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [X] Are the changes visual?
  - [X] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/locales/de/egg.ts [hunk 1]
```diff
@@ -17,5 +17,8 @@ export const egg: SimpleTranslationEntries = {
   "notEnoughVouchers": "Du hast nicht genug Ei-Gutscheine!",
   "tooManyEggs": "Du hast schon zu viele Eier!",
   "pull": "Pull",
-  "pulls": "Pulls"
+  "pulls": "Pulls",
+  "hatchFromTheEgg": "{{pokemonName}} hatched from the egg!",
+  "eggMoveUnlock": "Egg Move unlocked",
+  "rareEggMoveUnlock": "Rare Egg Move unlocked",
```
Comment (CodeTappert) [inline src/locales/de/egg.ts:23]:
```suggestion
  "hatchFromTheEgg": "Ein {{pokemonName}} ist aus dem Ei geschlüpft!",
  "eggMoveUnlock": "Ei-Attacke freigeschaltet",
  "rareEggMoveUnlock": "Seltene Ei-Attacke freigeschaltet",
```

File: src/locales/zh_CN/egg.ts [hunk 2]
```diff
@@ -17,5 +17,8 @@ export const egg: SimpleTranslationEntries = {
   "notEnoughVouchers": "你没有足够的兑换券！",
   "tooManyEggs": "你的蛋太多啦！",
   "pull": "次",
-  "pulls": "次"
+  "pulls": "次",
+  "hatchFromTheEgg": "{{pokemonName}} hatched from the egg!",
+  "eggMoveUnlock": "Egg Move unlocked",
+  "rareEggMoveUnlock": "Rare Egg Move unlocked",
```
Comment (40chyan) [inline src/locales/zh_CN/egg.ts:23]:
```suggestion
  "hatchFromTheEgg": "{{pokemonName}} 从蛋中孵化了！",
  "eggMoveUnlock": "蛋招式已解锁",
  "rareEggMoveUnlock": "稀有蛋招式已解锁",
```

**Comment by EnochG1** [comment]
image
image
@CodeTappert Please approve if test results are right.

**Comment by EnochG1** [comment]
image
image
@40chyan Please approve if test results are right.

**Comment by CodeTappert** [comment]
> image image @40chyan Please approve if test results are right.

The cursor position here looks pretty wrong

**Comment by EnochG1** [comment]
> The cursor position here looks pretty wrong

@CodeTappert 
image
I checked other case in game.
It might be not because of egg hatched message.

**Comment by 40chyan** [comment]
> > The cursor position here looks pretty wrong
> 
> @CodeTappert image I checked other case in game. It might be not because of egg hatched message but chinese exclusive.

yes exactly what I want to say. a Chinese string problem maybe @EnochG1

**Review by 40chyan — APPROVED** [review]

File: src/system/game-data.ts [hunk 3]
```diff
@@ -1551,7 +1552,8 @@ export class GameData {
       this.starterData[speciesId].eggMoves |= value;
 
       this.scene.playSound("level_up_fanfare");
-      this.scene.ui.showText(`${eggMoveIndex === 3 ? "Rare " : ""}Egg Move unlocked: ${allMoves[speciesEggMoves[speciesId][eggMoveIndex]].name}`, null, () => resolve(true), null, true);
+
+      this.scene.ui.showText(`${eggMoveIndex === 3 ? i18next.t("egg:rareEggMoveUnlock") : i18next.t("egg:eggMoveUnlock")}: ${allMoves[speciesEggMoves[speciesId][eggMoveIndex]].name}`, null, () => resolve(true), null, true);
```
Comment (hibiyasleep) [inline src/system/game-data.ts:1556]: I know this is a bit late, but can move name be included into translatable part?

> Egg Move unlocked: Harden
> 알 기술 해금: 단단해지기

would be better with

> Egg Move unlocked: _Harden_!
> 알 기술 _단단해지기_가 해금되었다! (…를 사용할 수 있게 되었다! maybe)

Comment (EnochG1) [reply src/system/game-data.ts]: Thanks for your advice! I think it's good idea.
좋은 생각인거 같아서 추가로 작업했습니다.
아래 코멘트에 테스트 스크린샷 첨부드릴테니 괜찮다 생각되면 approve 부탁드리겠습니다.

**Comment by EnochG1** [comment]
I approved @hibiyasleep's suggestion and make moveName to parameter in egg move unlock messages.
This is test screenshot of Korean and English(Not changed but for checking it works.)

image
image

File: src/locales/ko/egg.ts [hunk 4]
```diff
@@ -17,5 +17,8 @@ export const egg: SimpleTranslationEntries = {
   "notEnoughVouchers": "바우처가 충분하지 않습니다!",
   "tooManyEggs": "알을 너무 많이 갖고 있습니다!",
   "pull": "뽑기",
-  "pulls": "뽑기"
+  "pulls": "뽑기",
+  "hatchFromTheEgg": "알이 부화해서\n{{pokemonName}}[[가]] 태어났다!",
+  "eggMoveUnlock": "알 기술 {{moveName}}[[를]]\n사용할 수 있게 되었다.",
+  "rareEggMoveUnlock": "레어 알 기술 {{moveName}}[[를]]\n사용할 수 있게 되었다.",
```
Comment (hibiyasleep) [inline src/locales/ko/egg.ts:23]:
```suggestion
  "eggMoveUnlock": "알 기술 {{moveName}}[[를]]\n사용할 수 있게 되었다!",
  "rareEggMoveUnlock": "레어 알 기술 {{moveName}}[[를]]\n사용할 수 있게 되었다!",
```
to be consistent

Comment (EnochG1) [reply src/locales/ko/egg.ts]: Nice advice, thanks :)

File: src/locales/it/egg.ts [hunk 5]
```diff
@@ -17,5 +17,8 @@ export const egg: SimpleTranslationEntries = {
   "notEnoughVouchers": "Non hai abbastanza Biglietti!",
   "tooManyEggs": "Hai troppe Uova!",
   "pull": "Tiro",
-  "pulls": "Tiri"
+  "pulls": "Tiri",
+  "hatchFromTheEgg": "{{pokemonName}} hatched from the egg!",
```
Comment (EnochG1) [inline src/locales/it/egg.ts:21]:
```suggestion
  "hatchFromTheEgg": "Dall’Uovo è nato {{pokemonName}}!",
```

**Review by hibiyasleep — APPROVED** [review]

**Comment by Adri1** [comment]
@EnochG1 Do not forget the French translation too, submitted a bit earlier in this conversation 👌

**Comment by EnochG1** [comment]
@Adri1
I'm sorry but can you give me a link of your PR?
I can't find in my repo and I submitted your other PRs but that isn't this work.
https://github.com/EnochG1/pokerogue/pulls?q=is%3Aopen+is%3Apr

**Comment by Adri1** [comment]
@EnochG1 Not a PR, but a pending review in this very conversation 10+ hours ago

**Comment by EnochG1** [comment]
@Adri1 I can't find your review or suggestion.
image
image

Maybe I'm missing something.
Or if your review or suggestion is 'Pending', you have to submit your review at here.
image

**Comment by Adri1** [comment]
@EnochG1 What... Wierd because when I scroll up I still can see it.
Well I'm gonna make a new one

File: src/locales/fr/egg.ts [hunk 6]
```diff
@@ -17,5 +17,8 @@ export const egg: SimpleTranslationEntries = {
   "notEnoughVouchers": "Vous n’avez pas assez de coupons !",
   "tooManyEggs": "Vous avez trop d’Œufs !",
   "pull": "Tirage",
-  "pulls": "Tirages"
+  "pulls": "Tirages",
+  "hatchFromTheEgg": "{{pokemonName}} hatched from the egg!",
+  "eggMoveUnlock": "Egg Move unlocked: {{moveName}}",
+  "rareEggMoveUnlock": "Rare Egg Move unlocked: {{moveName}}",
```
Comment (Adri1) [inline src/locales/fr/egg.ts:23]:
```suggestion
  "hatchFromTheEgg": "{{pokemonName}} sort de l’Œuf !",
  "eggMoveUnlock": "Capacité Œuf débloquée ",
  "rareEggMoveUnlock": "Capacité Œuf Rare débloquée ",
```

Comment (Adri1) [reply src/locales/fr/egg.ts:23]: French Translation for these additions

Spaces added at the end of "eggMoveUnlock" and "rareEggMoveUnlock" sentences are completly normal.

File: src/locales/fr/egg.ts [hunk 7]
```diff
@@ -17,5 +17,8 @@ export const egg: SimpleTranslationEntries = {
   "notEnoughVouchers": "Vous n’avez pas assez de coupons !",
   "tooManyEggs": "Vous avez trop d’Œufs !",
   "pull": "Tirage",
-  "pulls": "Tirages"
+  "pulls": "Tirages",
+  "hatchFromTheEgg": "{{pokemonName}} sort de l’Œuf !",
+  "eggMoveUnlock": "Capacité Œuf débloquée ",
+  "rareEggMoveUnlock": "Capacité Œuf Rare débloquée ",
```
Comment (EnochG1) [inline src/locales/fr/egg.ts:23]:
```suggestion
  "eggMoveUnlock": "Capacité Œuf débloquée {{moveName}}",
  "rareEggMoveUnlock": "Capacité Œuf Rare débloquée {{moveName}}",
```

Comment (EnochG1) [reply src/locales/fr/egg.ts:23]: changed logic suggested by [hibiyasleep](https://github.com/hibiyasleep) must set moveName so I added it

**Comment by EnochG1** [comment]
@Adri1 I approved your suggestion and tested.
image
image
But second test result is not right.

After I fixed logic with Egg Move Unlocked, test result is below.
image

Please check test screenshot and approve when it is right!

File: src/locales/fr/egg.ts [hunk 8]
```diff
@@ -17,5 +17,8 @@ export const egg: SimpleTranslationEntries = {
   "notEnoughVouchers": "Vous n’avez pas assez de coupons !",
   "tooManyEggs": "Vous avez trop d’Œufs !",
   "pull": "Tirage",
-  "pulls": "Tirages"
+  "pulls": "Tirages",
+  "hatchFromTheEgg": "{{pokemonName}} sort de l’Œuf !",
+  "eggMoveUnlock": "Capacité Œuf débloquée {{moveName}}",
+  "rareEggMoveUnlock": "Capacité Œuf Rare débloquée {{moveName}}",
```
Comment (Adri1) [inline src/locales/fr/egg.ts:23]:
```suggestion
  "eggMoveUnlock": "Capacité Œuf débloquée : {{moveName}}",
  "rareEggMoveUnlock": "Capacité Œuf Rare débloquée : {{moveName}}",
```

**Review by Adri1 — COMMENTED** [review]
Oops I hav'nt noticed that ": {{moveName}}" has been added, sorry, I readapteded it
Except this tiny issue, everything is okay

File: src/locales/fr/egg.ts [hunk 9]
```diff
@@ -17,5 +17,8 @@ export const egg: SimpleTranslationEntries = {
   "notEnoughVouchers": "Vous n’avez pas assez de coupons !",
   "tooManyEggs": "Vous avez trop d’Œufs !",
   "pull": "Tirage",
-  "pulls": "Tirages"
+  "pulls": "Tirages",
+  "hatchFromTheEgg": "{{pokemonName}} sort de l’Œuf !",
+  "eggMoveUnlock": "Capacité Œuf débloquée : {{moveName}}",
+  "rareEggMoveUnlock": "Capacité Œuf Rare débloquée : {{moveName}}",
```
Comment (Adri1) [inline src/locales/fr/egg.ts:23]:
```suggestion
  "eggMoveUnlock": "Capacité Œuf débloquée :\n{{moveName}}",
  "rareEggMoveUnlock": "Capacité Œuf Rare débloquée :\n{{moveName}}",
```

**Review by Adri1 — COMMENTED** [review]
A really tini tiny edit, just to be really sure

**Review by Adri1 — APPROVED** [review]

**Comment by EnochG1** [comment]
@Adri1 Don't mind. I committed your suggestion :)
image

**Comment by Adri1** [comment]
Perfection 👌

File: src/locales/es/egg.ts [hunk 10]
```diff
@@ -17,5 +17,8 @@ export const egg: SimpleTranslationEntries = {
   "notEnoughVouchers": "¡No tienes suficientes vales!",
   "tooManyEggs": "¡No tienes suficiente espacio!",
   "pull": "Tirada",
-  "pulls": "Tiradas"
+  "pulls": "Tiradas",
+  "hatchFromTheEgg": "{{pokemonName}} hatched from the egg!",
+  "eggMoveUnlock": "Egg Move unlocked: {{moveName}}",
+  "rareEggMoveUnlock": "Rare Egg Move unlocked: {{moveName}}",
```
Comment (GoldTra) [inline src/locales/es/egg.ts:23]:
```suggestion
  "hatchFromTheEgg": "¡Ha salido un {{pokemonName}} del Huevo!",
  "eggMoveUnlock": "Mov. Huevo desbloqueado:\n{{moveName}}",
  "rareEggMoveUnlock": "Mov. Huevo Raro desbloqueado:\n{{moveName}}",
```

**Review by GoldTra — CHANGES_REQUESTED** [review]
Spanish Translations

**Review by GoldTra — APPROVED** [review]

**Comment by EnochG1** [comment]
@GoldTra Thanks, I committed and added test screenshot at main article :)

**Review by CodeTappert — APPROVED** [review]

**Review by returntoice — APPROVED** [review]

File: src/locales/zh_TW/egg.ts [hunk 11]
```diff
@@ -19,4 +19,7 @@ export const egg: SimpleTranslationEntries = {
   "pull": "抽",
   "pulls": "抽",
   "sameSpeciesEgg": "{{species}} 會從這個蛋裡孵化！"
```
Comment (EnochG1) [inline src/locales/zh_TW/egg.ts:21]:
```suggestion
  "sameSpeciesEgg": "{{species}} 會從這個蛋裡孵化！",
```

**Review by Tempo-anon — APPROVED** [review]

**Pull request merged**